test/extended/networking/service_network_consistent.go

```diff
@@ -0,0 +1,60 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/openshift/client-go/operatorcontrolplane/clientset/versioned/typed/operatorcontrolplane/v1alpha1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = g.Describe("[sig-network][Late] service network access from openshift-apiserver to kube-apiserver", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("service-network-openshift-apiserver")
+	)
+
+	g.It("shouldn't report outage to kubernetes-service", func() {
+		ctx := context.TODO()
+		endpointCheckClient := v1alpha1.NewForConfigOrDie(oc.AdminConfig())
+		connectivityChecks, err := endpointCheckClient.PodNetworkConnectivityChecks("openshift-apiserver").List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		failures := []string{}
+		for _, check := range connectivityChecks.Items {
+			if !strings.Contains(check.Name, "kube-service") {
+				continue
+			}
+			for _, outage := range check.Status.Outages {
+				failures = append(failures, fmt.Sprintf("%#v", outage))
+			}
+		}
+		if len(failures) > 0 {
+			g.Fail(fmt.Sprintf("the KUBERNETES_SERVICE_HOST:KUBERNETES_SERVICE_PORT was inaccessible via the service network IP (compare against kube-apiserver direct endpoint access):\n%v", strings.Join(failures, "\n")))
```

Comment:
`KUBERNETES_SERVICE_HOST` *is* `the service network IP`. (ie, this is basically saying "172.30.0.1 was inaccessible via 172.30.0.1").

Also, a random developer looking at test failures will have no idea what "(compare against kube-apiserver direct endpoint access)" is supposed to mean. Is that an instruction to the reader? If so, where is the reader supposed to get that other information, and why can't the test case make the comparison itself?

Also, you can use `g.Errorf(...)` instead of `g.Fail(fmt.Sprintf(...))`

Reply:
> Also, a random developer looking at test failures will have no idea what "(compare against kube-apiserver direct endpoint access)" is supposed to mean. Is that an instruction to the reader? If so, where is the reader supposed to get that other information, and why can't the test case make the comparison itself?

I'll see about adding more information. Basically we are seeing cases today where direct access to a kube-apiserver via the node IP works fine.  But access via 172.30.0.1 failed.  So we know the kube-apiservers (all of them) are accepting connections, but 172.30.0.1 shows repeated "connection refused"

Reply:
so I want to be sure no one bounces these as "well the kube-apiserver is returning connection refused".  In every case we've seen so far, the kube-apiserver is functioning, but cannot be accessed via the service network on a node that is reporting the network as ready.